diff --git a/lib/rules/forward-ref-uses-ref.js b/lib/rules/forward-ref-uses-ref.js
index 3a0b7de4c951e7ea92a9de7bac924ba95d643342..b1cfb74a59eaa05fa862b27d28f148f7d120bb2f 100644
--- a/lib/rules/forward-ref-uses-ref.js
+++ b/lib/rules/forward-ref-uses-ref.js
@@ -57,7 +57,7 @@ module.exports = {
   },
 
   create(context) {
-    const sourceCode = context.getSourceCode();
+    const sourceCode = context.sourceCode || (typeof context.getSourceCode === 'function' ? context.getSourceCode() : undefined);
 
     return {
       'FunctionExpression, ArrowFunctionExpression'(node) {
diff --git a/lib/rules/jsx-filename-extension.js b/lib/rules/jsx-filename-extension.js
index 4a2cfeede66b68255ca48bc38c0eb2f1537a785a..ae0d5cc71ffdbf4f2425f15dd0df4426d3e0f866 100644
--- a/lib/rules/jsx-filename-extension.js
+++ b/lib/rules/jsx-filename-extension.js
@@ -61,7 +61,7 @@ module.exports = {
   },
 
   create(context) {
-    const filename = context.getFilename();
+    const filename = context.filename || context.getPhysicalFilename ? context.getPhysicalFilename() : (context.getFilename ? context.getFilename() : '');
 
     let jsxNode;
 
diff --git a/lib/util/version.js b/lib/util/version.js
index c9dc683b7d149c5a3a941bdcbd32f657facdfbf6..00b5ce2586c7c225cde0b458932482f3a8dea34e 100644
--- a/lib/util/version.js
+++ b/lib/util/version.js
@@ -28,7 +28,7 @@ function resetDetectedVersion() {
 
 function resolveBasedir(contextOrFilename) {
   if (contextOrFilename) {
-    const filename = typeof contextOrFilename === 'string' ? contextOrFilename : contextOrFilename.getFilename();
+    const filename = typeof contextOrFilename === 'string' ? contextOrFilename : (typeof contextOrFilename.getFilename === 'function' ? contextOrFilename.getFilename() : contextOrFilename.filename);
     const dirname = path.dirname(filename);
     try {
       if (fs.statSync(filename).isFile()) {
